llama_index/chat_engine/types.py

```diff
@@ -29,6 +31,7 @@ class StreamingAgentChatResponse:
 
     response: str = ""
     sources: List[ToolOutput] = field(default_factory=list)
+    source_nodes: List[NodeWithScore] = field(default_factory=list)
```

Comment:
tbh, maybe this would be more helpful if it was an `@property` for both the agent response classes.

The property function could search `self.sources` for query engine sources, and return all source nodes.

Then it would work for everything, not just the condense query engine right?

Let me know what you think :)

Reply:
Yes that would be more generic. So iterate through the sources and check if there is a raw_output.source_nodes?

Thank YOU (the royal) for creating this great package. Happy to contribute.

Reply:
yea you got it, basically iterate and check for `isinstance(raw_output, RESPONSE_TYPE)`